sycl/source/detail/platform_impl.cpp

```diff
@@ -305,13 +306,17 @@ static void filterDeviceFilter(vector_class<RT::PiDevice> &PiDevices,
   int InsertIDx = 0;
   int DeviceNum = 0;
   for (RT::PiDevice Device : PiDevices) {
-    // Since device type is not determined yet, we use string search for
-    // a known word in the device name information.
-    string_class DeviceName =
-        sycl::detail::get_device_info<string_class, info::device::name>::get(
-            Device, Plugin);
-    std::transform(DeviceName.begin(), DeviceName.end(), DeviceName.begin(),
-                   ::tolower);
+    RT::PiDeviceType PiDevType;
+    info::device_type DeviceType = info::device_type::all;
+    Plugin.call<PiApiKind::piDeviceGetInfo>(Device, PI_DEVICE_INFO_TYPE,
+                                            sizeof(RT::PiDeviceType),
+                                            &PiDevType, nullptr);
+    if (PiDevType == PI_DEVICE_TYPE_ACC)
+      DeviceType = info::device_type::accelerator;
+    else if (PiDevType == PI_DEVICE_TYPE_CPU)
+      DeviceType = info::device_type::cpu;
+    else if (PiDevType == PI_DEVICE_TYPE_GPU)
+      DeviceType = info::device_type::gpu;
```

Comment:
NIT: device_type enum values are 1:1 with values of PiDeviceType(see sycl/include/CL/sycl/info/info_desc.hpp:147), so we could just cast device_type to PiDeviceType here.

Reply:
Thanks. Changed as you suggested.